### PR TITLE
feat(api): Upgrade Jobs & Report APIs to Version 2

### PR DESCRIPTION
**Pull request by dvjsharma**
## Description

This PR aims to upgrade the `/jobs` and `/report` endpoints to v2 in accordance with the new upgrade guidelines.

### Changes

* Completed all tasks mentioned in #2724 .
* Updated `openapiv2.yaml` to have all documentation changes.
* Modified models & unit tests to make them version aware.

### How to Test

* Open the `openapiv2.yaml` file in Swagger Editor and test all the updated endpoints for v2 specific responses. All function as intended.
* Execute unit tests, all pass without any issues.

CC @GMishx @shaheemazmalmmd @dushimsam

### CODE REVIEW

**Comment by dvjsharma** [comment]
Note: Merge after #2711

File: src/www/ui/api/Controllers/JobController.php [hunk 1]
```diff
@@ -148,16 +159,25 @@ public function getJobs($request, $response, $args)
...
+    }
     if (is_numeric($folder) && is_numeric($upload) && $folder > 0 && $upload > 0) {
       $scanOptionsJSON = $this->getParsedBody($request);
       if (empty($scanOptionsJSON)) {
         throw new HttpBadRequestException("No agents selected!");
       }
       $uploadHelper = new UploadHelper();
       $info = $uploadHelper->handleScheduleAnalysis($upload, $folder,
-        $scanOptionsJSON);
+        $scanOptionsJSON, false, $apiVersion);
```
Comment (dushimsam) [inline src/www/ui/api/Controllers/JobController.php:180]: Any reason for changing this?

File: src/www/ui/api/Models/Analysis.php [hunk 2]
```diff
@@ -103,14 +103,14 @@ public function __construct($bucket = false, $copyright = false, $ecc = false, $
...
-  public function setUsingArray($analysisArray)
+  public function setUsingArray($analysisArray, $version = ApiVersion::V1)
   {
     if (array_key_exists("bucket", $analysisArray)) {
       $this->bucket = filter_var($analysisArray["bucket"],
         FILTER_VALIDATE_BOOLEAN);
     }
-    if (array_key_exists("copyright_email_author", $analysisArray)) {
-      $this->copyright = filter_var($analysisArray["copyright_email_author"],
+    if (array_key_exists(($version == ApiVersion::V1? "copyright_email_author" : "copyrightEmailAuthor"), $analysisArray)) {
```
Comment (dushimsam) [inline src/www/ui/api/Models/Analysis.php:112]: Could we reverse the logic, the V1 should be the default as we're normally doing in controllers.
 
`($version == ApiVersion::V2 ? "copyrightEmailAuthor":"copyright_email_author")`

File: src/www/ui/api/Models/Analysis.php [hunk 3]
```diff
@@ -103,14 +103,14 @@ public function __construct($bucket = false, $copyright = false, $ecc = false, $
...
+      $this->copyright = filter_var($analysisArray[$version == ApiVersion::V1? "copyright_email_author" : "copyrightEmailAuthor"],
```
Comment (dushimsam) [inline src/www/ui/api/Models/Analysis.php:113]: Let API V1 be the default

File: src/www/ui/api/Models/Decider.php [hunk 4]
```diff
@@ -60,22 +60,22 @@ public function __construct($nomosMonk = false, $bulkReused = false, $newScanner
    * @param array $deciderArray Associative boolean array
    * @return Decider Current object
    */
-  public function setUsingArray($deciderArray)
+  public function setUsingArray($deciderArray, $version = ApiVersion::V1)
   {
-    if (array_key_exists("nomos_monk", $deciderArray)) {
-      $this->nomosMonk = filter_var($deciderArray["nomos_monk"],
+    if (array_key_exists(($version == ApiVersion::V1? "nomos_monk" : "nomosMonk"), $deciderArray)) {
```
Comment (dushimsam) [inline src/www/ui/api/Models/Decider.php:65]: Let's make the API V1 be the default

File: src/www/ui/api/Models/Decider.php [hunk 5]
```diff
@@ -60,22 +60,22 @@ public function __construct($nomosMonk = false, $bulkReused = false, $newScanner
...
+      $this->nomosMonk = filter_var($deciderArray[$version == ApiVersion::V1? "nomos_monk" : "nomosMonk"],
         FILTER_VALIDATE_BOOLEAN);
     }
-    if (array_key_exists("bulk_reused", $deciderArray)) {
-      $this->bulkReused = filter_var($deciderArray["bulk_reused"],
+    if (array_key_exists(($version == ApiVersion::V1? "bulk_reused" : "bulkReused"), $deciderArray)) {
```
Comment (dushimsam) [inline src/www/ui/api/Models/Decider.php:69]: Let's make the API V1 be the default

File: src/www/ui/api/Models/Decider.php [hunk 6]
```diff
@@ -60,22 +60,22 @@ public function __construct($nomosMonk = false, $bulkReused = false, $newScanner
...
+      $this->bulkReused = filter_var($deciderArray[$version == ApiVersion::V1? "bulk_reused" : "bulkReused"],
         FILTER_VALIDATE_BOOLEAN);
     }
-    if (array_key_exists("new_scanner", $deciderArray)) {
-      $this->newScanner = filter_var($deciderArray["new_scanner"],
+    if (array_key_exists(($version == ApiVersion::V1? "new_scanner" : "newScanner"), $deciderArray)) {
```
Comment (dushimsam) [inline src/www/ui/api/Models/Decider.php:73]: Let's make the API V1 be the default

File: src/www/ui/api/Models/Decider.php [hunk 7]
```diff
@@ -60,22 +60,22 @@ public function __construct($nomosMonk = false, $bulkReused = false, $newScanner
...
+      $this->newScanner = filter_var($deciderArray[$version == ApiVersion::V1? "new_scanner" : "newScanner"],
```
Comment (dushimsam) [inline src/www/ui/api/Models/Decider.php:74]: Let's make the API V1 be the default

File: src/www/ui/api/Models/Decider.php [hunk 8]
```diff
@@ -60,22 +60,22 @@ public function __construct($nomosMonk = false, $bulkReused = false, $newScanner
...
         FILTER_VALIDATE_BOOLEAN);
     }
-    if (array_key_exists("ojo_decider", $deciderArray)) {
-      $this->ojoDecider = filter_var($deciderArray["ojo_decider"],
+    if (array_key_exists(($version == ApiVersion::V1? "ojo_decider" : "ojoDecider"), $deciderArray)) {
```
Comment (dushimsam) [inline src/www/ui/api/Models/Decider.php:77]: Let's make the API V1 be the default

File: src/www/ui/api/Models/Reuser.php [hunk 9]
```diff
@@ -81,25 +81,25 @@ public function __construct($reuseUpload, $reuseGroup, $reuseMain = false,
    * @throws \UnexpectedValueException If reuse upload of reuse group are non
    * integers
    */
-  public function setUsingArray($reuserArray)
+  public function setUsingArray($reuserArray, $version = ApiVersion::V1)
```
Comment (dushimsam) [inline src/www/ui/api/Models/Reuser.php:84]: Let's make sure the API V1 is the default, in the implementations of this method.

File: src/www/ui/api/Controllers/JobController.php (hunk 1 above)
Comment (dvjsharma) [reply src/www/ui/api/Controllers/JobController.php:180]: PTAL at https://github.com/fossology/fossology/pull/2736/files#diff-8a7e9933b8879b6ff5e18b4082859fccababcfdde56142701d2e83b298aa4a5cR115
If I don't pass `false`, `$newUpload` will take up the `$apiVersion`.

File: src/www/ui/api/Models/Analysis.php (hunk 2 above)
Comment (dvjsharma) [reply src/www/ui/api/Models/Analysis.php:112]: Done

File: src/www/ui/api/documentation/openapiv2.yaml [hunk 10]
```diff
@@ -2815,7 +2815,7 @@ paths:
     parameters:
       - name: groupName
         description: The group name to chose while accessing jobs
-        in: header
+        in: qyery
```
Comment (GMishx) [inline src/www/ui/api/documentation/openapiv2.yaml:2818]:
```suggestion
        in: query
```

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor fix needed.

**Comment by GMishx** [comment]
And as https://github.com/fossology/fossology/pull/2711 is merged, you can rebase the branch.

File: src/www/ui/api/documentation/openapiv2.yaml (hunk 10 above)
Comment (dvjsharma) [reply src/www/ui/api/documentation/openapiv2.yaml]: Done

**Pull request merged**